lib/puppet/type/hocon_setting.rb

```diff
@@ -1,70 +1,190 @@
+# frozen_string_literal: true
+
 Puppet::Type.newtype(:hocon_setting) do
   ensurable do
     defaultvalues
     defaultto :present
   end
 
   newparam(:setting, namevar: true) do
-    desc 'The name of the setting to be defined.'
+    desc <<~DESC
+      The name of the HOCON file setting to be defined.
+
+      This can be a top-level setting or a setting nested within another
+      setting.
+
+      To define a nested setting, give the full path to that setting
+      with each level separated by a `.` So, to define a setting `foosetting`
+      nested within a setting called `foo` contained on the top level, the
+      `setting` parameter would be set to `foo.foosetting`.
+
+      This parameter, along with `path`, is one of two namevars for the
+      `hocon_setting` type, meaning that Puppet will give an error if two
+      `hocon_setting` resources have the same `setting` and `path` parameters.
+
+      If no `setting` value is explicitly set, the title of the resource will
+      be used as the value of `setting`.
+    DESC
   end
 
   newparam(:path, namevar: true) do
-    desc 'The file Puppet will ensure contains the specified setting.'
+    desc <<~DESC
+      The HOCON file in which Puppet will ensure the specified setting.
+
+      This parameter, along with `setting`, is one of two namevars for the
+      `hocon_setting` type, meaning that Puppet will give an error if two
+      `hocon_setting` resources have the same `setting` and `path` parameters.
+    DESC
+
     validate do |value|
-      unless (Puppet.features.posix? && value =~ %r{^\/}) || (Puppet.features.microsoft_windows? && (value =~ %r{^.:\/} || value =~ %r{^\/\/[^\/]+\/[^\/]+}))
-        raise(Puppet::Error, "File paths must be fully qualified, not '#{value}'")
-      end
+      raise(Puppet::Error, "File paths must be fully qualified, not '#{value}'") unless (Puppet.features.posix? && value =~ %r{^/}) || (Puppet.features.microsoft_windows? && (value =~ %r{^.:/} || value =~ %r{^//[^/]+/[^/]+}))
```

Comment:
This is worse in readability. The guard cop while the length cop is disabled makes things worse.

Reply:
There should be a CopCop for things like this. 😛